pkg/assembler/backends/neo4j/builder.go

```diff
@@ -38,3 +47,58 @@ func (bn builderNode) PropertyNames() []string {
 func (bn builderNode) IdentifiablePropertyNames() []string {
 	return []string{"uri"}
 }
+
+func (c *neo4jClient) Builders(ctx context.Context, builderSpec *model.BuilderSpec) ([]*model.Builder, error) {
+	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
+	defer session.Close()
+
+	result, err := session.ReadTransaction(
+		func(tx neo4j.Transaction) (interface{}, error) {
+
+			var sb strings.Builder
+			var firstMatch bool = true
+			queryValues := map[string]any{}
+
+			sb.WriteString("MATCH (n:Builder)")
+
+			if builderSpec.URI != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "n", "uri", "$builderUri")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
```

Comment:
missing setting `firstMatch=false`, i see what you're doing with having the blocks be modular, but for most cases, it looks like it isn't quite used. maybe remove it if its deterministic and then use it only for the for loops?

Reply:
so the linter errored on that saying it will never be used. If you see the execution it's true that even if firstMatch is set it won't be used after that point. `firstMatch` is used to determine if a `WHERE` is needed or if the `WHERE` exists, than use `AND` to append another condition.

Reply:
Depending on what is specified in the query. So depending on where the query start, a `WHERE` is needed, and in other cases `AND`

Reply:
ah yes you are right. Removed where its not needed.